test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py

```diff
@@ -1212,6 +1212,14 @@ def _validate_argument_spec(self, docs, spec, kwargs, context=None):
                     code='parameter-alias-repeated',
                     msg=msg
                 )
+            data_choices = data.get('choices', set())
+            if not context and arg == 'state' and data_choices:
```

Comment:
Could as well drop this now:
```suggestion
            if not context and arg == 'state':
```
because the intersection with an empty set is always an empty set which will result in no iterations of that for-loop